pkg/operator/ceph/cluster/osd/health.go

```diff
@@ -66,20 +70,24 @@ func (m *Monitor) Start(stopCh chan struct{}) {
 
 // OSDStatus validates osd dump output
 func (m *Monitor) osdStatus() error {
-	logger.Debugf("OSDs with previously detected Down status: %+v", m.lastStatus)
+	logger.Debugf("OSDs with previously detected Down status: %+v", m.lastDownStatus)
+	logger.Debugf("OSDs with previously detected Out status: %+v", m.lastOutStatus)
 	osdDump, err := client.GetOSDDump(m.context, m.clusterName)
 	if err != nil {
 		return err
 	}
 	logger.Debugf("osd dump %v", osdDump)
 
-	evalDownStatus := func(id int) {
-		if now := time.Now(); now.Sub(m.lastStatus[id]) > osdGracePeriod {
-			logger.Warningf("osd.%d has been down for longer than the grace period (down since %+v)", id, m.lastStatus[id])
-			m.lastStatus[id] = time.Now()
+	//returns true if the laststatus is more than osd grace period else returns false
+	evalOutOrDownStatus := func(id int, state string, lastStatus map[int]time.Time) bool {
```

Comment:
returning a `bool` value in order to take action if the osdGracePeriod is exceeded for `down` or out` osds